tests/unit/test_install.py

```diff
@@ -542,3 +542,47 @@ def test_validate_source_dir(tmp_run_dir: Callable, tmp_src_dir: Callable):
     with pytest.raises(WorkflowFilesError) as exc_info:
         validate_source_dir(src_dir, 'ajay')
     assert "exists in source directory" in str(exc_info.value)
+
+
+def test_install_workflow_failif_name_name(tmp_src_dir):
+    """If a run_name is given validate_workflow_name is called on
+    the workflow and the run name in combination.
+    """
+    src_dir: Path = tmp_src_dir('ludlow')
+    # It only has a workflow name:
+    with pytest.raises(WorkflowFilesError, match='can only contain'):
+        install_workflow(src_dir, workflow_name='foo?')
+    # It only has a run name:
+    with pytest.raises(WorkflowFilesError, match='can only contain'):
+        install_workflow(src_dir, run_name='foo?')
+    # It has a legal workflow name, but an invalid run name:
+    with pytest.raises(WorkflowFilesError, match='can only contain'):
+        install_workflow(src_dir, workflow_name='foo', run_name='bar?')
```

Comment:
With these sorts of tests where we are testing for failure, it is often a good idea to test that the thing you are testing can actually succeed when the conditions are right.

This avoids the situation where the test setup is broken causing the action to fail with the expected error, but for the wrong reasons (though the `match` kwarg provides some protection against this).